From b1f41d52dd6eb44b298bbd6b5e5fd60f310ccddf Mon Sep 17 00:00:00 2001
From: Riddhi Aggarwal <quic_riddagga@quicinc.com>
Date: Tue, 22 Aug 2023 10:57:16 +0530
Subject: [PATCH] mtd: rawnand: qcom: Fix serial training logic for all
 possible no.of trained phases

In qspi training, the case with no.of successfully trained phases 3 is not handled properly.
Added a fix to handle this case.

Change-Id: I5361144f79bcbffd040ca0b48dcb6e15961cceaa
Signed-off-by: Riddhi Aggarwal <quic_riddagga@quicinc.com>
---
 drivers/mtd/nand/raw/qcom_nandc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mtd/nand/raw/qcom_nandc.c b/drivers/mtd/nand/raw/qcom_nandc.c
index d4932806f35e..f6718a6aca8d 100644
--- a/drivers/mtd/nand/raw/qcom_nandc.c
+++ b/drivers/mtd/nand/raw/qcom_nandc.c
@@ -3371,6 +3371,8 @@ static int qspi_get_appropriate_phase(struct qcom_nand_controller *nandc, u8 *ph
 	}
 
 	/* Filter out middle phase */
+	if (cnt == 1)
+		phase = phase_table[1];
 	if (cnt > 1 && cnt <= TOTAL_NUM_PHASE) {
 		if (!(cnt & 1))
 			phase = phase_ranges[cnt/2 - 1];
-- 
2.34.1

